hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/contracts/ContractsModule.java

```diff
@@ -185,8 +188,10 @@ static ContractValidationRule providePrefixCodeRule() {
     static MessageCallProcessor provideV_0_30MessageCallProcessor(
             final @V_0_30 EVM evm,
             final @V_0_30 PrecompileContractRegistry precompiles,
-            final Map<String, PrecompiledContract> hederaPrecompileList) {
-        return new HederaMessageCallProcessor(evm, precompiles, hederaPrecompileList);
+            final Map<String, PrecompiledContract> hederaPrecompileList,
```

Comment:
For replay and version consistency I don't think we should make these changes to 0.30 and 0.34. They should be left as is to not change the responses of the system at these points.
These changes should only be for 0.38.0 EVM.
Same for all the other cases